bcolz/ctable.py

```diff
@@ -516,7 +516,13 @@ def delcol(self, name=None, pos=None):
             name = self.names[pos]
 
         # Remove the column
-        self.cols.pop(name)
+        col = self.cols.pop(name)
+
+        # remove the data if we have a rootdir
```

Comment:
I am thinking the carray needs to know how to delete itself. Seperation of concerns.